webapp-django/crashstats/manage/views.py

```diff
@@ -1,78 +1,33 @@
-from django.core.urlresolvers import reverse
-from django.db import connection, transaction
-from django.shortcuts import get_object_or_404, redirect, render
-from django.utils import timezone
-from django.views.decorators.http import require_POST
-
-from pinax.eventlog.models import log, Log
+from django.db import connection
+from django.shortcuts import redirect, render
+
 import requests
 
-from crashstats.crashstats.models import (
-    GraphicsDevices,
-    Reprocessing,
-)
+from crashstats.crashstats.models import GraphicsDevices
 from crashstats.supersearch.models import SuperSearchMissingFields
-from crashstats.tokens.models import Token
-from crashstats.status.models import StatusMessage
 from crashstats.crashstats.utils import json_view
 from crashstats.manage.decorators import superuser_required
 
 from . import forms
 from . import utils
 
 
-def notice_change(before, after):
-    assert before.__class__ == after.__class__
-    changes = {}
-    if isinstance(before, User) or isinstance(before, Group):
-        for field in before._meta.get_fields():
-            fieldname = getattr(field, 'attname', field.name)
-            v1 = getattr(before, fieldname, None)
-            v2 = getattr(after, fieldname, None)
-            if hasattr(v1, 'all'):
-                # many-to-many field!
-                # To be able to compare, the many-to-many field needs to
-                # have been converted to a list and attached to the object.
-                # If we don't do this, we won't notice the difference.
-                # Remember that many-to-many fields are stored in a different
-                # table. E.g. for a User:
-                #
-                #  user_id | group_id
-                #  --------|---------
-                #  3       | 45
-                #  3       | 89
-                #
-                # And all you have is a User instance with the ID 3,
-                # you can't find out what was in that many-to-many mapping
-                # table before because now it's 45 and 89.
-                # So it must have been expanded into a Python list and
-                # attached to the object itself.
-                if not hasattr(before, '__%s' % fieldname):
-                    continue
-                # these have to have been expanded before!
-                v1 = getattr(before, '__%s' % fieldname)
-
-                v2 = getattr(
-                    after,
-                    '__%s' % fieldname,
-                    [unicode(x) for x in v2.all()]
-                )
-
-            if v1 != v2:
-                changes[fieldname] = [v1, v2]
-        return changes
-    raise NotImplementedError(before.__class__.__name__)
+def log_action(user_id, action_flag, change_message):
+    LogEntry.objects.create(
+        user_id=user_id,
+        content_type_id=None,
+        object_id=None,
+        object_repr='',
+        action_flag=action_flag,
+        change_message=change_message
+    )
```

Comment:
Previously, we logged actions with the pinax eventlog module. This changes things so we're logging actions using the Django admin LogEntry thing. It's definitely not as ergonomic and there's like no documentation for it. The `LogEntryManager` has a helper method on it, but requires you to pass in an `object_id` and `content_type_id`, so it doesn't work for us.

So... that's why this is here.

Pretty sure the only thing we need it for is the graphics devices stuff. I want to move that table over to the Django admin. Once I do that, then we can manage it with the admin and won't need this weirdo code. I've never added graphics devices things. So maybe we should just nix this now and not worry about it?

Reply:
Hrm. I think I just convinced myself it's a good idea to just nix it--less to deal with, less to test.

Honk your horn twice and retweet if you think that's a bad idea.